http-client/src/main/java/io/micronaut/http/client/netty/ConnectionManager.java

```diff
@@ -306,8 +305,18 @@ private static NioEventLoopGroup createEventLoopGroup(HttpClientConfiguration co
      * @see DefaultHttpClient#start()
      */
     public void start() {
-        group = createEventLoopGroup(configuration, threadFactory);
-        bootstrap.group(group);
+        // only need to start new group if it's managed by us
+        if (shutdownGroup) {
+            group = createEventLoopGroup(configuration, threadFactory);
+            initBootstrap(); // rebuild bootstrap with new group
```

Comment:
Do we need to do some cleanup of `bootstrap` if it already exists at this point?

Reply:
nah it doesn't hold any resources